test/test_core.py

```diff
@@ -6095,6 +6105,10 @@ def test_unistd_io(self):
       self.emcc_args = orig_compiler_opts + ['-D' + fs]
       if fs == 'NODEFS':
         self.emcc_args += ['-lnodefs.js']
+        if config.NODE_JS not in config.JS_ENGINES:
+          # NODEFS requires node
+          continue
```

Comment:
Why is this needed here but not in the diff above it that just uses `require_node`?

Reply:
`require_node` will call `skipTest` it can't be used to skip over just one loop iteration like this;  It would skip the entire rest of the test.

Ideally we would avoid loops in tests for this reason, but I wasn't able convert this test to parameterized form.